monai/utils/nvtx.py

```diff
@@ -53,12 +55,26 @@ def __init__(
         name: Optional[str] = None,
         methods: Optional[Union[str, Tuple[str, ...]]] = None,
         append_method_name: Optional[bool] = None,
+        recursive: bool = False,
     ) -> None:
         self.name = name
         self.methods = methods
         self.append_method_name = append_method_name
+        self.recursive = recursive
 
     def __call__(self, obj: Any):
+        if self.recursive is True:
+            if isinstance(obj, (list, tuple)):
+                return type(obj)(Range(recursive=True)(t) for t in obj)
+
+            from monai.transforms.compose import Compose
+
+            if isinstance(obj, Compose):
+                obj.transforms = Range(recursive=True)(obj.transforms)
+                return Range(self.name, recursive=False)(obj)
```

Comment:
thanks, looks like this line created an additional `Range` object which is not really needed? 
```suggestion
```